src/main/java/org/Olivia/EventDispatcher.java

```diff
@@ -0,0 +1,221 @@
+package org.Olivia;
+/**
+ * Listens command from a given UiHandler
+ * parse the command and execute the corresponding function
+ * syncs with hard disk after each execution
+ * @author ZHANG TONGJUN (albertZhangTJ)
+ */
+
+import org.Olivia.IO.FileHandler;
+import org.Olivia.IO.UiHandler;
+import org.Olivia.calendar.*;
+import java.io.File;
+import java.security.InvalidParameterException;
+import java.util.List;
+
+
+public class EventDispatcher {
+
+    private Calendar table;
+    private UiHandler ui;
+    private FileHandler disk;
+
+    public EventDispatcher(Calendar table, UiHandler ui, FileHandler disk) {
+        this.table=table;
+        this.ui=ui;
+        this.disk=disk;
```

Comment:
There should be a space between the variables in `this.disk=disk`! I think this can be applied to the rest of your code.